cmd/acmesolver/app/app.go

```diff
@@ -37,9 +38,15 @@ func NewACMESolverCommand(stopCh <-chan struct{}) *cobra.Command {
 			rootCtx = logf.NewContext(rootCtx, nil, "acmesolver")
 			log := logf.FromContext(rootCtx)
 
+			completedCh := make(chan struct{})
 			go func() {
+				defer close(completedCh)
 				<-stopCh
-				if err := s.Shutdown(rootCtx); err != nil {
+				// allow a timeout for graceful shutdown
+				ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+				defer cancel()
```

Comment:
This is a common pattern for http servers (see example https://pkg.go.dev/net/http#Server.Shutdown).
"Shutdown gracefully shuts down the server without interrupting any active connections. Shutdown works by first closing all open listeners, then closing all idle connections, and then waiting indefinitely for connections to return to idle and then shut down. If the provided context expires before the shutdown is complete, Shutdown returns the context's error, otherwise it returns any error returned from closing the Server's underlying Listener(s)."